llvm/include/llvm/CodeGen/SDPatternMatch.h

```diff
@@ -47,6 +47,8 @@ class BasicMatchContext {
   bool match(SDValue N, unsigned Opcode) const {
     return N->getOpcode() == Opcode;
   }
+
+  static constexpr bool IsVP = false;
```

Comment:
**mshockwave**: https://github.com/llvm/llvm-project/pull/102877#discussion_r1715581991

> have you explored the ideas of abstracting the number of operands instead? Specifically, having a match context to implement its own version of `unsigned getNumOperands(SDValue N)` (or something similar)?

Reply:
I'll try that.